src/VASSAL/counters/Labeler.java

```diff
@@ -154,7 +154,7 @@ public void mySetType(String type) {
    */
   protected String clean(String s) {
     // Cannot use $PieceName$ in a label format, must use $pieceName$
-    return s.replaceAll("$"+BAD_PIECE_NAME+"$", "$"+PIECE_NAME+"$");
+    return s.replace("$"+BAD_PIECE_NAME+"$", "$"+PIECE_NAME+"$");
```

Comment:
I see why this happened---having functions named `replace` and `replaceAll` suggests that the former will replace the first occurrence while the latter will replace all occurrences---but that's not what they do.